src/processor/mapper/map_extend.cpp

```diff
@@ -31,16 +31,21 @@ static std::vector<property_id_t> populatePropertyIds(
 }
 
 static std::unique_ptr<RelTableCollection> populateRelTableCollection(table_id_t boundNodeTableID,
-    const RelExpression& rel, RelDataDirection direction, const expression_vector& properties,
+    const RelExpression& rel, ExtendDirection extendDirection, const expression_vector& properties,
```

Comment:
I think we need to rewrite this function. 

`populateRelTableCollection` is used to collect storage data so it should be using `RelDataDirection`. But what we get from `logicalExtend` is `ExtendDirection`. So the right thing to do is to switch different cases of `ExtendDirection` and find its corresponding `RelDataDirection`. I'll give you a skeleton code.

```
static std::pair<DirectedRelTableData*, std::unique_ptr<RelTableScanState>>
getRelTableDataAndScanState(RelDataDirection direction) {
    if (schema->getBoundTableID(direction) != boundNodeTableID) {
        // No data stored for given direction and boundNode.
        return std::make_pair(nullptr, nullptr);
    }
    auto relData = relsStore.getRelTable(relTableID)->getDirectedTableData(direction);
    std::vector<property_id_t> propertyIds;
    for (auto& property : properties) {
        auto propertyExpression = reinterpret_cast<PropertyExpression*>(property.get());
        propertyIds.push_back(propertyExpression->hasPropertyID(relTableID) ?
                                  propertyExpression->getPropertyID(relTableID) :
                                  INVALID_PROPERTY_ID);
    }
    auto scanState = make_unique<RelTableScanState>(
        std::move(propertyIds), relsStore.isSingleMultiplicityInDirection(direction, relTableID) ?
                                    RelTableDataType::COLUMNS :
                                    RelTableDataType::LISTS);
    return std::make_pair(relData, std::move(scanState));
}

static std::unique_ptr<RelTableCollection> populateRelTableCollection(ExtendDirection extendDirection){
    std::vector<DirectedRelTableData*> tables;
    std::vector<std::unique_ptr<RelTableScanState>> tableScanStates;
    for (auto relTableID : rel.getTableIDs()) {
        auto relTableSchema = catalog.getReadOnlyVersion()->getRelTableSchema(relTableID);
        switch (extendDirection) {
        case common::ExtendDirection::FWD: { 
            auto [relTableData, scanState] = getRelTableDataAndScanState(RelDataDirection::FWD);
            // push into vectors.
        }
        case common::ExtendDirection::BWD: {
            auto [relTableData, scanState] = getRelTableDataAndScanState(RelDataDirection::BWD);
            // push into vectors.
        }
        case common::ExtendDirection::BOTH: {
            auto [relTableData, scanState] = getRelTableDataAndScanState(RelDataDirection::FWD);
            // push into vectors.
            auto [relTableData, scanState] = getRelTableDataAndScanState(RelDataDirection::BWD);
            // push into vectors.
        }
        }
    }
    return std::make_unique<RelTableCollection>(std::move(tables), std::move(tableScanStates));
}
```

`populateRelTableCollection` handles the switch between `ExtendDirection` and `RelDataDirection`. 
While `getRelTableDataAndScanState` fetch storage data based on `RelDataDirection`. You might notice `getRelTableDataAndScanState` is the exact piece of code we had before.